src/storage/mod.rs

```diff
@@ -477,12 +477,23 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
     }
 
     // TODO: refactor to use `Api` parameter.
-    fn check_api_version_ranges(
+    fn check_api_version_ranges<I, K>(
         storage_api_version: ApiVersion,
         req_api_version: ApiVersion,
         cmd: CommandKind,
-        ranges: impl IntoIterator<Item = (Option<impl AsRef<[u8]>>, Option<impl AsRef<[u8]>>)>,
-    ) -> Result<()> {
+        ranges: I,
+        reverse: bool,
+    ) -> Result<()>
+    where
+        I: IntoIterator<Item = (Option<K>, Option<K>)>,
+        K: AsRef<[u8]>,
+    {
+        let mut ranges: Box<dyn iter::Iterator<Item = (Option<K>, Option<K>)>> =
```

Comment:
It seems too complicated. Shall we just check the first item for vector of ranges ?
Please also refer to https://github.com/tikv/tikv/issues/12207.

Reply:
The simplification might be placed into another pull request. I think this pull request should preserve the previous behaviour of the TiKV server.